pkg/hooks/hooks.go

```diff
@@ -137,9 +137,9 @@ func (h *VolumeManagerHooks) Update(oldObj, newObj interface{}) {
 	if newVolumeManager.Status.State == states.Failed {
 		// Delete all the sub-resources.
 		for _, handler := range h.dataHandlers {
-			for _, vConfig := range newVolumeManager.Spec.VolumeConfigs {
+			for idx, vConfig := range newVolumeManager.Spec.VolumeConfigs {
 				if handler.GetSourceType() == vConfig.SourceType {
-					handler.OnDelete(newVolumeManager.Namespace, vConfig, *controllerRef)
+					handler.OnDelete(newVolumeManager.Namespace, vConfig, newVolumeManager.Status.Volumes[idx], *controllerRef)
```

Comment:
Would this ever cause an index error? I would say adding an error handling here would help. WDYT?

Reply:
I don't think it will cause an index error because we add vstatuses = # of vc even if the CR fails. https://github.com/kubeflow/experimental-kvc/blob/master/pkg/hooks/hooks.go#L76-L80. 

There are issues that can pop-up specific to each type. Those need to be handled behind the interface. Example: https://github.com/balajismaniam/experimental-kvc/blob/ee7cc18b241b97b1781977a4d9a67127fdd731c8/pkg/handlers/s3_handler.go#L233.

Reply:
Sounds good.

Reply:
This won't also be called when the source type is not registered due to the `if` block.